### DISCUSSION TITLE
List of entities not refresing on Delete [Cache]

### DISCUSSION THREAD

**Discussion by HadiSDev** [urql-graphql/urql#2834] [Q&A] [answered]
I am fairly new to Urql. I am trying to use the Cache Exchange. Specifically I am trying to update a list of Entities when a delete mutation is executed. First the queries (I am using Codegen):

Query for the list of entities:
```
query getEntities($created_by: UUID!){
    entitiesCollection(filter: {created_by: {eq: $created_by}}) {

        edges {
            node {
                id,
                ... <--- other fields
                __typename
            }
        }
    }
}
```
Delete Mutation:
```
mutation DeleteEntity($id: UUID!){
    deleteFromEntitiesCollection(filter: {id: {eq: $id}}, atMost: 1){
        records {
            __typename,
            id
        }
    }
}
```

Cache exhange:
```
exchanges: [
        dedupExchange,
        cacheExchange({}),
        fetchExchange,
      ],
```

React Side where list is loaded:
```
  const [query] = useGetEntitiesQuery({
    variables: { created_by: user?.id },
    requestPolicy: "cache-and-network",
  });
```

React Side where entity is deleted:
`const [_, deleteEntity] = useDeleteEntityMutation();`

Right now the data only updates locally on refresh but I want it to update without any manual refreshes.

What am I doing wrong?

**Comment by JoviDeCroock** [comment, marked as answer]
We can't predict what deletes/creates do to a list hence you need a custom updater https://formidable.com/open-source/urql/docs/graphcache/cache-updates/

That for instance does cache.invalidate or updateQuery

**Reply by HadiSDev** [reply]
I see. I tried the example from the tutorial given:
```
deleteEntity(_result, args, cache, _info) {
  console.log("Delete called");
  cache.invalidate({
    __typename: "entity",
    id: args.id as string,
  });
},
```

I am a bit confused on why the update is not triggered :/ Since "console.log("Delete called");" is not called I assume it is because my setup of in the cache context is wrong with the "deleteEntity(_result, args, cache, _info)" signature.

**Reply by JoviDeCroock** [reply]
I mean 😅 you did not really link the error... The signature should be

```
cacheExchange({
  updates: { Mutation: { NameOfYourMutation: () => {} } }
})
```

**Reply by HadiSDev** [reply]
Sorry the warning sentence was not supposed to be there. There is no warning :) Is the NameOfYourMutation case sensitive?

**Reply by JoviDeCroock** [reply]
Yes, everything in terms of identifiers, typenames, ... are case-sensitive. Just for clarity, an OperationName isn't the name of your mutation-field

**Reply by HadiSDev** [reply]
so if this is my mutation:
```
mutation DeleteEntity($id: UUID!){
    deleteFromEntitiesCollection(filter: {id: {eq: $id}}, atMost: 1){
        records {
            __typename,
            id
        }
    }
}
```

The mutation name is "DeleteEntity"?

**Reply by JoviDeCroock** [reply]
no it's `deleteFromEntitiesCollection` - DeleteEntity is the name of your operation, where the operation can hold multiple fields like `deleteFromEntitiesCollection` and more

**Reply by HadiSDev** [reply]
Thanks, it works now.

**Comment by kitten** [comment]
Small question, I see that this issue came down to you using "operation names" (which are basically just names you give to an operation — like untyped, free-form code comments), rather than the fields in the normalised cache. I know this may be hard to explain, but do you recollect what made you confuse the two? We've gotten this issue several times now, and are trying to figure out whether a specific part of the documentation accidentally causes any confusion here, which we could fix. :v:

**Reply by HadiSDev** [reply]
Yes, when I read the documentation I thought it was not case-sensitive in the name and that was why you had named it "RemoveTodo" in the query:
![billede](https://user-images.githubusercontent.com/10367624/204138540-d098c638-c414-4503-8abb-b1c368ac6987.png)

since that did not work for me I tried to name it after the operation name but I accidentally named it in the wrong case.  

I think what could maybe change is just make it in bold so that it is case-sensitive and that you have to use the operation name (I know you are already doing that but just more ). Perhaps an illustration that demonstrates the difference between operation names and mutation name, similar to the illustration on: https://formidable.com/open-source/urql/docs/basics/document-caching/
